### PR TITLE
Task sizing: Add more dimensions to prometheus metrics

### PR DESCRIPTION
**Pull request by bduffany**
---

**Version bump**: Patch

### CODE REVIEW

**Review by vadimberezniker — APPROVED** [review]
I'm fine with these changes, but I know @tylerwilliams has previously expressed concerns about using group ID as a dimension since it can cause these metrics to have high cardinality. I personally think the trade-off is worth it. Please get his LGTM before merging.

**Comment by bduffany** [comment]
For more context, I plan on using the group ID label to check whether certain groups are running into errors with task sizer misses more often than other groups, and then that will allow looking at the actions for those groups to see what is going on.

**Comment by bduffany** [comment]
@vadimberezniker I am going to merge this to get it in this week's release -- I don't expect that the cardinality will be an issue for this metric since we already have other metrics using group ID. If we need to optimize metric size later then we can always revisit.

**Pull request merged**
